pallets/anchor-handler/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+use pallet_anchor::types::AnchorInterface;
+use sp_std::prelude::*;
+
+pub use pallet::*;
+
+use pallet_bridge::types::{ChainId, ResourceId};
+pub mod types;
+use types::*;
+
+type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::{dispatch::DispatchResultWithPostInfo, pallet_prelude::*};
+	use frame_system::pallet_prelude::*;
+	use pallet_anchor::types::{AnchorInspector, AnchorMetadata, EdgeMetadata};
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	/// The module configuration trait.
+	pub trait Config: frame_system::Config + pallet_anchor::Config + pallet_bridge::Config {
+		/// The overarching event type.
+		type Event: IsType<<Self as frame_system::Config>::Event> + From<Event<Self>>;
+
+		/// Specifies the origin check provided by the bridge for calls that can
+		/// only be called by the bridge pallet
+		type BridgeOrigin: EnsureOrigin<Self::Origin, Success = Self::AccountId>;
+
+		/// The currency mechanism.
+		type Currency: Currency<Self::AccountId>;
+
+		/// Anchor Interface
+		type Anchor: AnchorInterface<Self> + AnchorInspector<Self>;
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn maintainer)]
+	/// The parameter maintainer who can change the parameters
+	pub(super) type Maintainer<T: Config> = StorageValue<_, T::AccountId, ValueQuery>;
+
+	/// The map of trees to their anchor metadata
+	#[pallet::storage]
+	#[pallet::getter(fn anchors)]
+	pub type Anchors<T: Config> = StorageMap<_, Blake2_128Concat, ResourceId, T::TreeId, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn update_records)]
+	/// sourceChainID => height => Update Record
+	pub type UpdateRecords<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::ChainId,
+		Blake2_128Concat,
+		T::BlockNumber,
+		UpdateRecord<T::AccountId, ResourceId, T::ChainId, T::Element, T::BlockNumber>,
+		ValueQuery,
+	>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn counts)]
+	/// The number of updates
+	pub(super) type Counts<T: Config> = StorageMap<_, Blake2_128Concat, T::ChainId, u64, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	#[pallet::metadata(T::AccountId = "AccountId")]
+	pub enum Event<T: Config> {
+		MaintainerSet(T::AccountId, T::AccountId),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Access violation.
+		InvalidPermissions,
+		/// Storage overflowed.
+		StorageOverflow,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_finalize(_n: BlockNumberFor<T>) {
+			// FIXME create/init anchors
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(195_000_000)]
+		pub fn set_maintainer(origin: OriginFor<T>, new_maintainer: T::AccountId) -> DispatchResultWithPostInfo {
```

Comment:
Is maintainer being used for anything here? usually this was for pallets with parameters. If not we can remove.